### PR TITLE
Shell escape

### PR DESCRIPTION
**Pull request by csgillespie**

### CODE REVIEW

**Comment by jamieRowen** [comment]
@jwalton3141 TRAVIS checks still required to pass before review

**Comment by sebble** [comment]
You should be able to run `inteRgrate::check_via_env(default = TRUE)` to do these checks locally.

Also, as a reviewer, I read about the security implications. What are your reasons for enabling this globally?

**Comment by jwalton3141** [comment]
@sebble We are attempting to use the `ifplatform` package. This package requires shell-escape enabled to work. The `ifplatform` package should either be removed, if it is superfluous, or allowed to work properly with shell-escape.

**Comment by jamieRowen** [comment]
@jwalton3141 Why are we using the `ifplatform` package? Also when looking at the merge conflicts, I have to admit I don't understand what your change does

**Comment by jwalton3141** [comment]
@jamieRowen I'm not sure why the `ifplatform` package is included. I assumed it was included for a good reason, though maybe I am wrong on this count.

Compiling with `ifplatform` and without `shell-escape` produces an error / warning, although compilation still completes. I do not feel strongly about `ifplatform` vs `shell-escape`, but I don't like ignoring errors.

**Comment by jamieRowen** [comment]
@csgillespie do you know why we are using `ifplatform`? part of CI?

**Comment by sebble** [comment]
If it's only for Windows we could perhaps catch and ignore the warning. Otherwise here's a discussion: https://tex.stackexchange.com/questions/64233/how-do-i-determine-what-items-to-add-to-my-restricted-shell-escape-safe-list

**Comment by csgillespie** [comment]
I've no idea what ifplatform is. Can we look at the history and see when it was added?

**Comment by csgillespie** [comment]
Also I think this merge has went wrong. For example.check the version in the master description.  Possibly easier deleting and trying again

**Comment by jamieRowen** [comment]
@jwalton3141 I've sent you a couple of messages in slack, but till copy them in here for reference,

```
im trying to get to the bottom of the ifplatform thing, your message states we are attempting to use it, unless I am being thick I can't see where we are trying to use it

the only place it turns up in a grep is your news.md addition

the only other place that string shows up is in the git log of messages
```

Also your branch is behind master, can you merge in the changes before we can look at the PR. There are a couple of merge conflicts too, although they look like they should be fairly easy to resolve.

**Comment by jwalton3141** [comment]
Package ifplatform Warning: 
    shell escape is disabled, so I can only detect \ifwindows.

**Comment by jwalton3141** [comment]
If I run `make` in the `notes/` directory of `jrIntroduction_python_notes`, I see the `ifplatform` warning during compilation. If I check `main.log` I can see the warning there too:
```
Package ifplatform Warning: 
    shell escape is disabled, so I can only detect \ifwindows.
```

**Comment by jamieRowen** [comment]
@sebble are we worried about the potential security implications, on the assumption that `ifplatform` comes from a dep,

@jwalton3141 hate to be the bearer of bad news, but you are failing the TRAVIS checks again

**Comment by sebble** [comment]
@jamieRowen I just want to consider the security implications, that's why we have reviews.. I don't know the reason for requiring shell (Jack only added it to hide a warning), nor the reason for ifplatform (if more than Windows, heed that warning), nor the impact of allowing arbitrary executable code in our LaTeX docs.. it's up to us..

**Comment by jamieRowen** [comment]
Presumably that warning has been present for some time then. Do we need to stop the warning? Or do we just trap or allow it some other way?

**Comment by jamieRowen** [comment]
I'm leaning on the side of we don't just want to allow shell escape

File: DESCRIPTION [hunk 1]
```diff
@@ -46,5 +46,4 @@ Suggests:
 VignetteBuilder: 
     knitr
 Encoding: UTF-8
-Roxygen: list(markdown = TRUE)
```
Comment (csgillespie) [inline DESCRIPTION:49]: Why has this been removed?

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # jrNotes 0.6.2 _2020-08-10_
   * Bug fix: check condition for whether there is news
+  * Bug: Allow shell-escape compilation for ifplatform package
```
Comment (csgillespie) [inline NEWS.md:3]: This won't work. You'll need to bump the version and create a new NEWS item

**Comment by csgillespie** [comment]
Closing this. `ifplatform` is part of (I think) `tufte`. Although it emits a warning, it shouldn't affect anyone as we (sort of) enforce xelatex.

**Pull request closed without merging**